otp4gb/parameters.py

```diff
@@ -162,7 +168,20 @@ def _calculate_distance_matrix(
         gpd.GeoSeries(distances["destination_centroid"])
     )
 
-    return distances.set_index(["origin", "destination"])["distance"]
+    same_zone = distances["origin"] == distances["destination"]
+    flagged = (same_zone & (distances["distance"] > FLAG_DISTANCE_INTERNAL)).sum()
```

Comment:
This seems to flag potential trips if the calculated crow-fly distance exceeds 10km (`FLAG_DISTANCE_INTERNAL = 10_000` assumed metres)

`FLAG_DISTANCE_INTERNAL` is a local constant, could it be a config variable? There may be use cases where a lot of trips would exceed this (i.e., longer range car & bus)

Additionally, `flagged` contains the count of `same_zone` trips. This is then reported to the user with `LOG.warning` below as counting as a trip that exceeds the `FLAG_DISTANCE_INTERNAL`, when it could just be a same zone trip under the `FLAG_DISTANCE_INTERNAL` distance.

I don't know the motive behind the %of `flagged` trips that are `same_zone` trips (that is `{flagged / same_zone.sum():.1%}`), but reporting on the separate instances may be clearer?

 I.e.,
```
LOG.warning(
            f"{flagged:,} flagged trips detected. {same_zone.sum()} same zone trips"
            f"detected and {flagged - same_zone.sum()} internal crow-fly distance"
            f" > {FLAG_DISTANCE_INTERNAL:,}"
        )
```

Apologies if I've miss-read & understood this here.

Reply:
I think it's a good point about changing it to a config variable that is always handy.

I wanted to clear up that this code is only checking the distance between the same zone i.e. flagging zones where the origin centroid is more than 10,000km from the destination centroid. This is a check added due to there being an error in the employment weighted centroids data I used which was difficult to find, it won't flag 2 different zones.

The flagged array is only True for zones with the same origin and destination ID and a distance > 10,000km
`flagged = (same_zone & (distances["distance"] > FLAG_DISTANCE_INTERNAL)).sum()`

Hope this makes it a bit clearer

Reply:
Thank you for the clarification, think I'd been looking at a screen too long as this does make sense to me now. Bit of a brain fart where I was interpreting & as or 😅. 

Additionally, hadn't fully remembered distances are converted into km ( / 1000 somewhere) so the FLAG_DISTANCE_INTERNAL is a check for 10,000km, not 10,000m

Reply:
Actually those distances haven't been converted to km yet they're still in metres. The flag is actually 10km like you originally said. That's still good for flagging the issues I was looking but I think it's definately worth allowing it to be a parameter that can be changed, I'm going to create an issue suggesting the change